query-request.hh

```diff
@@ -282,16 +305,29 @@ public:
         : cf_id(std::move(cf_id))
         , schema_version(std::move(schema_version))
         , slice(std::move(slice))
-        , row_limit(static_cast<uint32_t>(row_limit))
+        , row_limit_low_bits(static_cast<uint32_t>(row_limit))
         , timestamp(now)
         , trace_info(std::move(ti))
         , partition_limit(static_cast<uint32_t>(partition_limit))
         , query_uuid(query_uuid)
         , is_first_page(is_first_page)
         , max_result_size(max_result_size)
+        , row_limit_high_bits(static_cast<uint32_t>(static_cast<uint64_t>(row_limit) >> 32))
         , read_timestamp(rt)
     { }
 
+
+    uint64_t get_row_limit() const {
+        return (static_cast<uint64_t>(row_limit_high_bits.value_or(0)) << 32) | row_limit_low_bits;
+    }
+    void set_row_limit(uint32_t new_row_limit) {
+        row_limit_low_bits = new_row_limit;
+        row_limit_high_bits = 0;
+    }
```

Comment:
Is this overload needed? the other overload will work just fine.

Reply:
I think I wanted to avoid excessive casts, removed in the next push